### PR TITLE
Problem: tx log attribtue value not parsable by some client

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #614

Solution:
- encode the value to json string rather than bytes

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: rpc/ethereum/backend/utils.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package backend
 import (
 	"bytes"
 	"errors"
+	"json"
```
Comment (yihuang) [inline rpc/ethereum/backend/utils.go:6]:
```suggestion
	"encoding/json"
```

File: x/evm/keeper/msg_server.go [hunk 2]
```diff
@@ -3,6 +3,7 @@ package keeper
 import (
 	"context"
 	"fmt"
+	"json"
```
Comment (yihuang) [inline x/evm/keeper/msg_server.go:6]:
```suggestion
	"encoding/json"
```

**Review by Piyushblog — COMMENTED** [review]
It is good but somewhere it is wrong

**Comment by thomas-nguy** [comment]
related to this issue https://github.com/informalsystems/tendermint-rs/issues/1003

**Review by thomas-nguy — APPROVED** [review]

File: rpc/ethereum/backend/utils.go [hunk 3]
```diff
@@ -147,7 +148,9 @@ func TxLogsFromEvents(codec codec.Codec, events []abci.Event) []*ethtypes.Log {
 			}
 
 			var log evmtypes.Log
-			codec.MustUnmarshal(attr.Value, &log)
+			if err := json.Unmarshal(attr.Value, &log); err != nil {
+				panic(err)
```
Comment (fedekunze) [inline rpc/ethereum/backend/utils.go:152]: ah I wouldn't panic here, let's return an error

**Review by fedekunze — CHANGES_REQUESTED** [review]
LGTM. Only left 1 comment. Let's also add a changelog entry 👍

File: rpc/ethereum/backend/utils.go [hunk 4]
```diff
@@ -147,7 +148,9 @@ func TxLogsFromEvents(codec codec.Codec, events []abci.Event) []*ethtypes.Log {
 			}
 
 			var log evmtypes.Log
-			codec.MustUnmarshal(attr.Value, &log)
+			if err := json.Unmarshal(attr.Value, &log); err != nil {
```
Comment (thomas-nguy) [inline rpc/ethereum/backend/utils.go:151]: codec is unused

File: rpc/ethereum/backend/utils.go (hunk 3 above)
Comment (fedekunze) [reply rpc/ethereum/backend/utils.go:152]: bump

**Comment by fedekunze** [comment]
please add a changelog entry too so that we can track the bug-fixes

**Review by fedekunze — APPROVED** [review]

**Pull request merged**